src/mongo_client.ts

```diff
@@ -500,6 +501,16 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
 
       Promise.all(activeSessionEnds)
         .then(() => {
+          // If we would attempt to select a server and get nothing back we short circuit
+          // to avoid the server selection timeout.
+          const selector = readPreferenceServerSelector(ReadPreference.primaryPreferred);
+          const topologyDescription = this.topology.description;
```

Comment:
Can `this.topology` be undefined here? If so, what would be the correct way to handle that?

Reply:
I've add the check after to address this.